backend/services/grading_embedding_service.py

```
@@ -0,0 +1,388 @@
+"""
+Enhanced Embedding Service for Grading Materials RAG
+Implements semantic chunking for optimal retrieval of grading criteria, rubrics, and references
+"""
+
+import os
+import openai
+import asyncio
+import hashlib
+from typing import List, Dict, Tuple, Optional
```

Ruff (0.14.1)

`typing.List` is deprecated, use `list` instead

(UP035)